govcd/api.go

```diff
@@ -41,6 +38,10 @@ func (cli *Client) NewRequest(params map[string]string, method string, reqUrl ur
 	// Add the params to our URL
 	reqUrl.RawQuery = reqValues.Encode()
 
+	for key, value := range notEncodeParams {
```

Comment:
1. Do we need to check whether `value` is empty?
2. What if `key` is empty?
3. What if `rawQuery` is empty? We'd get an incorrect query then (starting with `&`)

Reply:
Added check to verify value and key, also starting with & is ok. Verified.